src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemorySegment.java

```diff
@@ -330,24 +331,22 @@ default MemorySegment asSlice(long offset) {
      * this case, or when no overlap occurs, {@code null} is returned.
      *
      * @param other the segment to test for an overlap with this segment.
-     * @return a new memory segment, or {@code null} if no overlap occurs.
+     * @return a new slice of this segment, or {@code null} if no overlap occurs.
      */
-    MemorySegment overlap(MemorySegment other);
+    MemorySegment asOverlappingSlice(MemorySegment other);
 
     /**
      * Returns the offset, in bytes, from this segment to the other segment.
      *
      * <p>The offset is relative to the {@linkplain #address() base address} of
```

Comment:
Note that MemorySegment::address is only defined for native segments. Perhaps remove the link?